server/src/main/java/org/elasticsearch/http/HttpClientStatsTracker.java

```diff
@@ -226,7 +227,18 @@ synchronized void update(HttpRequest httpRequest, HttpChannel httpChannel, long
             lastRequestTimeMillis = currentTimeMillis;
             lastUri = httpRequest.uri();
             requestCount += 1;
-            requestSizeBytes += httpRequest.content().length();
+            if (httpRequest.content().isFull()) {
+                requestSizeBytes += httpRequest.content().asFull().bytes().length();
+            } else {
+                var contentStream = httpRequest.content().asStream();
+                var chunkHandler = contentStream.handler();
+                Consumer<HttpContent.Chunk> bytesCountingHandler = (chunk) -> {
+                    requestSizeBytes += chunk.bytes().length();
+                    chunkHandler.accept(chunk);
+                };
+                contentStream.setHandler(bytesCountingHandler);
```

Comment:
Is this going to repeatedly nesting the handler?

Reply:
No, we set handler once on prepareRequest. There is no recursion.